File: EmployeeDirectory/EmployeeDirectory.Data/DataModels/Role.cs
namespace EmployeeDirectory.Data.DataModels;

public partial class Role
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public int DepartmentId { get; set; }

    public string? Description { get; set; }

    public int LocationId { get; set; }
    public Location Location { get; set; } = null!;
    public Department Department { get; set; } = null!;
}

File: src/models/role.ts
export class Role{
    id:number;
    name:string;
    department:string;
    roleDescription:string;
    location:string;
    constructor(id:number, name:string, department:string, roleDescription:string, location:string){
        this.id=id;
        this.name=name;
        this.department=department;
        this.roleDescription=roleDescription;
        this.location=location;
    }
}